sphinx_gallery/notebook.py

```diff
@@ -142,36 +145,47 @@ def add_code_cell(work_notebook, code):
     work_notebook["cells"].append(code_cell)
 
 
-def add_markdown_cell(work_notebook, text):
+def add_markdown_cell(work_notebook, markdown):
     """Add a markdown cell to the notebook
 
     Parameters
     ----------
-    code : str
-        Cell content
+    markdown : str
+        Markdown cell content.
     """
     markdown_cell = {
         "cell_type": "markdown",
         "metadata": {},
-        "source": [rst2md(text)]
+        "source": [markdown]
     }
     work_notebook["cells"].append(markdown_cell)
 
 
-def fill_notebook(work_notebook, script_blocks):
+def fill_notebook(work_notebook, script_blocks, gallery_conf):
     """Writes the Jupyter notebook cells
 
+    If available, uses pypandoc to convert rst to markdown.
+
     Parameters
     ----------
     script_blocks : list
         Each list element should be a tuple of (label, content, lineno).
     """
-
     for blabel, bcontent, lineno in script_blocks:
         if blabel == 'code':
             add_code_cell(work_notebook, bcontent)
         else:
-            add_markdown_cell(work_notebook, bcontent + '\n')
+            if (isinstance(gallery_conf["pypandoc"], dict) and
+                    _has_pypandoc(raise_error=True)):
```

Comment:
```suggestion
            if isinstance(gallery_conf["pypandoc"], dict):
```
Then here you don't even need to check. (The way you have it coded now, it would emit a warning/log for every block of every example I think.)

Reply:
Eurgh yes I did it again. Was trying to avoid the logic at the top.